loader-tests/utils.ts

```diff
@@ -1,63 +1,91 @@
+import {
+  enableTestTime,
+  getTestEnvironment,
+  setBaseTestRoute,
+  setRequestMethod,
+  setupEnvironment,
+  userId
+} from '@slowreader/core'
 import type { ReadableAtom } from 'nanostores'
-import { isAbsolute, join, resolve } from 'node:path'
+import { readFile } from 'node:fs/promises'
+import { isAbsolute, join } from 'node:path'
 import pico from 'picocolors'
 
-const supportedExtensions = ['.xml', '.opml']
-
-export function isSupportedExt(path: string): boolean {
-  return supportedExtensions.some(ext => path.endsWith(ext))
+export async function readText(path: string): Promise<string> {
+  let absolute = path
+  if (!isAbsolute(absolute)) {
+    absolute = join(process.env.INIT_CWD!, path)
+  }
+  let buffer = await readFile(absolute)
+  return buffer.toString('utf-8')
 }
 
 export function isString(attr: null | string): attr is string {
   return typeof attr === 'string' && attr.length > 0
 }
 
-export function resolvePath(path: string): string {
-  if (isAbsolute(path)) {
-    return path
-  }
-  return resolve(join(process.cwd(), '..', path))
+export function enableTestClient(): void {
+  setupEnvironment(getTestEnvironment())
+  enableTestTime()
+  userId.set('10')
+  setBaseTestRoute({ params: {}, route: 'home' })
+  setRequestMethod(fetch)
 }
 
 export function waitForStoreResolve(
   store: ReadableAtom<boolean>
 ): Promise<boolean> {
-  return new Promise(_resolve => {
+  return new Promise(resolve => {
     let loadingComparison = 'idle'
     let unbind = store.listen(value => {
       if (value && loadingComparison === 'idle') {
         loadingComparison = 'loading'
       }
       if (!value && loadingComparison === 'loading') {
         unbind()
-        _resolve(value)
+        resolve(value)
       }
     })
   })
 }
 
-interface Logger {
-  err(msg: string): void
-  info(msg: string): void
-  log(msg: string): void
-  succ(msg: string): void
-  warn(msg: string): void
-}
-
-export const logger: Logger = {
-  err(msg) {
-    this.log(pico.red(`Error: ${msg}`))
-  },
-  info(msg) {
-    this.log(pico.blue(msg))
-  },
-  log(msg) {
-    process.stderr.write(`${msg}\n`)
-  },
-  succ(msg) {
-    this.log(pico.green(msg))
-  },
-  warn(msg) {
-    this.log(pico.yellow(`Warning: ${msg}`))
+interface NoFileError extends Error {
+  code: string
+  path: string
+}
+
+function isNoFileError(e: unknown): e is NoFileError {
+  return e instanceof Error && `code` in e && e.code === 'ENOENT'
+}
+
+export function print(msg: string): void {
```

Comment:
Logger separation is just to reduce a little LOC by avoid having separated types.

But it is very subjective.